cypress/e2e/smoke/balances.cy.js

```diff
@@ -0,0 +1,228 @@
+
+      // Select EUR
+      cy.get('ul[role="listbox"]').findByText('EUR').click()
+
+      // First row Fiat balance should not contain USDC
+      cy.get(balanceSingleRow).first().find('td').eq(FIAT_AMOUNT_COLUMN).should('not.contain', 'USDC')
+      // First row Fiat balance should contain EUR
+      cy.get(balanceSingleRow).first().find('td').eq(FIAT_AMOUNT_COLUMN).contains('€')
+    })
+  })
+
+  describe('pagination should work', () => {
+    before(() => {
+      // Open the Safe used for testing pagination
+      cy.visit(`/${PAGINATION_TEST_SAFE}/balances`, { failOnStatusCode: false })
+      cy.contains('button', 'Accept selection').click()
+
+      //Ensure the table is fully loaded
+      cy.contains('Rows per page:', { timeout: 10000 })
+    })
+    it('should should allow 25 rows per page and navigate to next and previous page', () => {
+      // Click on the pagination select dropdown
+      cy.get('[aria-haspopup="listbox"]').contains('25').click()
+
+      // Select 25 rows per page
+      cy.get('[aria-haspopup="listbox"]').get('li[role="option"]').contains('25').click()
+
+      // Table should have 25 rows
+      cy.contains('1–25 of 27')
+      cy.get(balanceSingleRow).should('have.length', 25)
+
+      // Click on the next page button
+      cy.get('button[aria-label="Go to next page"]').click()
+
+      // Table should have 1 rows
+      cy.contains('26–27 of 27')
+      cy.get(balanceSingleRow).should('have.length', 2)
+
+      // Click on the previous page button
+      cy.get('button[aria-label="Go to previous page"]').click()
+
+      // Table should have 25 rows
+      cy.contains('1–25 of 27')
+      cy.get(balanceSingleRow).should('have.length', 25)
+    })
+  })
+
+  describe('should open assets modals', () => {
+    const receiveAssetsModalTestId = 'div[aria-labelledby^=":r1"]'
+
+    it('should open the Receive assets modal', () => {
+      // Assets table container should exist
+      cy.get('[data-track="overview: Show Safe QR code"]').should('be.visible').click()
+
+      // The Receive Assets modal should be present
+      cy.get(receiveAssetsModalTestId).should('be.visible')
+
+      // Receive assets should be present
+      cy.get(receiveAssetsModalTestId).findByText('Receive assets').should('be.visible')
+
+      // The Receive screen should have the correct address
+      const [, safeAddress] = PAGINATION_TEST_SAFE.split(':')
+      cy.get(receiveAssetsModalTestId).findByText(safeAddress).should('be.visible')
+
+      // Checking and unchecking the QR code
+      cy.get(receiveAssetsModalTestId).find('[type="checkbox"]').should('be.checked')
+
+      cy.get(receiveAssetsModalTestId).find('[type="checkbox"]').uncheck().should('not.be.checked')
+
+      // Click in the Done button
+      cy.get(receiveAssetsModalTestId).find('[data-testid="CloseIcon"]').click()
+
+      // The Receive screen should be hidden
+      cy.get(receiveAssetsModalTestId).should('not.exist')
+    })
+  })
```

Comment:
this button does not exist on the Balances table. If you want to keep this test I suggest moving it out of this file.

Reply:
You can still open the modal on the "QR" code button on the side. If someday we make a test about the sidebar I'll move it there. I don't want to lose the steps.

Reply:
I suggest you create a test for the Sidebar Header then. This test here is misplaced